modules/authorization/src/utils/index.ts

```diff
@@ -31,56 +31,128 @@ export const computePermissionTuple = (
-        SELECT * FROM "cnd_ActorIndex"         
-        WHERE subject = '${subject}'
-      ) as actors
-      INNER JOIN (
-          SELECT * FROM "cnd_ObjectIndex"
-          WHERE subjectType = '${objectType}' AND subjectPermission = '${action}'
-      ) as obj 
-      ON actors.entity = obj.entity
-    )
-    UNION (
-      SELECT "computedTuple" 
-      FROM "cnd_Permission"          
-      WHERE "computedTuple" LIKE '${computedTuple}%'
-    )
-  ) idx
-  ON  idx.entity LIKE '%' || TEXT(s._id) || '%'
-  `;
+export interface AccessListQueryParams {
+  objectTypeCollection: string;
+  computedTuple: string;
+  subject: string;
+  objectType: string;
+  action: string;
 }
 
-export function getSQLAccessListQuery(
-  objectTypeCollection: string,
-  computedTuple: string,
-  subject: string,
-  objectType: string,
-  action: string,
-) {
-  return `SELECT ${objectTypeCollection}.* FROM ${objectTypeCollection}
+export function getPostgresAccessListQuery(params: AccessListQueryParams) {
+  const { objectTypeCollection, computedTuple, subject, objectType, action } = params;
+  return `SELECT "${objectTypeCollection}".* FROM "${objectTypeCollection}"
           INNER JOIN (
-              SELECT * FROM cnd_Permission
-              WHERE computedTuple LIKE '${computedTuple}%'
-          ) permissions ON permissions.computedTuple = '${computedTuple}:' || ${objectTypeCollection}._id
+              SELECT * FROM "cnd_Permission"
+              WHERE "computedTuple" LIKE '${computedTuple}%'
+          ) permissions ON permissions."computedTuple" = '${computedTuple}:' || "${objectTypeCollection}"._id
           INNER JOIN (
-              SELECT * FROM cnd_ActorIndex
+              SELECT * FROM "cnd_ActorIndex"
               WHERE subject = '${subject}'
           ) actors ON 1=1
           INNER JOIN (
-              SELECT * FROM cnd_ObjectIndex
-              WHERE subjectType = '${objectType}' AND subjectPermission = '${action}'
+              SELECT * FROM "cnd_ObjectIndex"
+              WHERE subject LIKE '${objectType}:%#${action}'
           ) objects ON actors.entity = objects.entity;`;
 }
+
+export function getSQLAccessListQuery(params: AccessListQueryParams) {
+  const { objectTypeCollection, computedTuple, subject, objectType, action } = params;
+  return `SELECT ${objectTypeCollection}.* FROM ${objectTypeCollection}
+        INNER JOIN (
+            SELECT * FROM cnd_Permission
+            WHERE computedTuple LIKE '${computedTuple}%'
+        ) permissions ON permissions.computedTuple = '${computedTuple}:' || ${objectTypeCollection}._id
+        INNER JOIN (
+            SELECT * FROM cnd_ActorIndex
+            WHERE subject = '${subject}'
+        ) actors ON 1=1
+        INNER JOIN (
+            SELECT * FROM cnd_ObjectIndex
+            WHERE subject LIKE '${objectType}:%#${action}'
+        ) objects ON actors.entity = objects.entity;`;
+}
+
+export function getMongoAccessListQuery(params: AccessListQueryParams) {
```

Comment:
I think this is the query we used to have a long time ago because it doesn't seem to match with the one deleted in createAccessList. Please check and update it we the new one we have if I'm correct